bids/layout/index.py

```diff
@@ -205,7 +205,12 @@ def index_metadata(self):
 
                 if ext == 'json':
                     with open(bf.path, 'r') as handle:
-                        payload = json.load(handle)
+                        try:
+                            payload = json.load(handle)
+                        except Exception as e:
+                            msg = ("Error occurred while trying to decode JSON"
+                                   " from file '{}'.".format(bf.path))
+                            raise Exception(msg) from e
```

Comment:
```suggestion
                            raise IOError(msg) from e
```

Or maybe `ValueError`?